src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.To.cs

```diff
@@ -37,8 +51,13 @@ public override string ToString()
                 }
             }
 
-            using PooledByteBufferWriter output = WriteToPooledBuffer(writerOptions: new JsonWriterOptions { Indented = true });
-            return JsonHelpers.Utf8GetString(output.WrittenMemory.Span);
+#pragma warning disable CA1869 // TODO https://github.com/dotnet/roslyn-analyzers/issues/6957
+            JsonSerializerOptions options =
+                s_indentedToStringOptions ??
+                Interlocked.CompareExchange(ref s_indentedToStringOptions, new JsonSerializerOptions { MaxDepth = JsonWriterOptions.DefaultMaxDepth, WriteIndented = true, }, null) ??
+                s_indentedToStringOptions;
```

Comment:
I wonder the overhead of `Lazy<T>` here. More overhead on startup even with `PublicationOnly`? Another level of indirection on every access?